src/lib/src/x86_arrops.cpp

```diff
@@ -0,0 +1,69 @@
+#include "arch_config.h"
+
+#ifdef DISTOPIA_X86_SSE4_1
+
+#include <cstddef>
+#include <immintrin.h>
+
+#include "arrops.h"
+#include "x86_swizzle.h"
+#include "ops.h"
+
+namespace {
+
+template<typename FloatingT, typename VectorT>
+void CalcBondsOrthoX86Vec(const FloatingT* coords0, const FloatingT* coords1,
+                          const FloatingT* box, std::size_t n,
+                          FloatingT* out) {
+  constexpr std::size_t vector_size = sizeof(VectorT) / sizeof(FloatingT);
```

Comment:
We can add a static type checks here 

```c++
static_assert(std::is_floating_point<FloatingT>::value, "Floating point required required.");
```
Not sure if there is an equivalent for VectorT (at least not stl)

Reply:
We can put a static check in there if you want, but I think it’d be redundant: if `FloatingT` does not define all required operations, the substitution will simply not compile. Plus, it would be nice to support _any_ type as long as those operations are defined (e.g. I may wish to invent a one-and-a-half-precision float; since it’s not native it won’t be recognised by `is_floating_point`).

Reply:
I take your point and am convinced. Isn't this also guarded by the std::enable_if?

Reply:
`CalcBondsOrthoX86Vec` specifically implements x86 vector type semantics so it doesn’t need `std::enable_if`: it is guarded by the C preprocessor.

Reply:
👍